### ISSUE TITLE
Multiple checkins occurring, not seeing Mayor status on checkin

### ISSUE THREAD

**Issue by GoogleCodeExporter** [iamakshayv/foursquared#148]
```
What steps will reproduce the problem?
1. Check-in on a venue, or Quick Check-in
2.
3.

What is the expected output? What do you see instead?
Twice now it's entered two check-ins for the venue, so the check-in
response dialog only shows "you just checked in at this venue ..." message.

The one time that it sent only one check-in the Mayor status was not
present on the dialog.

What version of the product are you using? On what operating system?
Android, 2009122101

Please provide any additional information below.

```

Original issue reported on code.google.com by `drat...@gmail.com` on 23 Dec 2009 at 7:33

**Comment by GoogleCodeExporter** [comment]
```
Here's my history from yesterday...


@Target - Cottage Grove (19 hours ago)
@Target - Cottage Grove (19 hours ago)
@Rainbow Foods - Woodbury (20 hours ago)
@Rainbow Foods - Woodbury (20 hours ago)
@Target - Woodbury (21 hours ago)
@Target - Woodbury (21 hours ago)
@Blissful Bath (21 hours ago)
@Blissful Bath (21 hours ago)
@Best Buy (21 hours ago)
@Best Buy (21 hours ago)

```

Original comment by `breonn...@gmail.com` on 23 Dec 2009 at 7:52

**Comment by GoogleCodeExporter** [comment]
```
What kind of phone do you have?
```

Original comment by `jlapenna` on 23 Dec 2009 at 10:45

**Comment by GoogleCodeExporter** [comment]
```
Moto Droid for me, Running Android 2.0.1 (Moto build ESD56)
```

Original comment by `drat...@gmail.com` on 23 Dec 2009 at 10:48

**Comment by GoogleCodeExporter** [comment]
```
I have a Sprint HTC Hero with Android 1.5.
```

Original comment by `breonn...@gmail.com` on 24 Dec 2009 at 12:27

**Comment by GoogleCodeExporter** [comment]
```
I had the double-checkin problem too on my G1, but it went away with the latest 
update 
(to 2009122300)
```

Original comment by `cakatz` on 24 Dec 2009 at 3:28

**Comment by GoogleCodeExporter** [comment]
```
fixed in 2009122300
live in the market since last night

```

Original comment by `naveen...@gmail.com` on 24 Dec 2009 at 6:17
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
I have 2010-03-28.  It seems to be happening on my Moto Droid running 2.1.  It 
doesn't automatically check in 
twice - but something about resuming into the app and navigating back to home 
screen to check in again has 
resulted check-ins at old places right before i check in at a new one.  I 
deleted all the instances from my history, 
but i'm having to do it every couple days and delete three or four check-ins.
```

Original comment by `jackson....@gmail.com` on 9 Apr 2010 at 4:50
